### PR TITLE
Assembly: Insert Component: Fix assembly not appearing

### PR DESCRIPTION
**Pull request by PaddleStroke**
Fix https://github.com/FreeCAD/FreeCAD/issues/14933

Fix another unlisted bug where groups where appearing even if they had nothing interesting (joint group were appearing for example). It was because it was using OutListRecursive and groundedObject linked to a part.
So instead it now use claimChildren. So the tree is building correctly.
This PR also adds 'claimChildrenRecursive' to view provider.

### CODE REVIEW

File: src/Gui/ViewProviderPyImp.cpp [hunk 1]
```diff
@@ -393,7 +393,23 @@ PyObject* ViewProviderPy::claimChildren(PyObject* args)
 
     std::vector<App::DocumentObject*> children = this->getViewProviderPtr()->claimChildren();
     Py::List ret;
-    for(App::DocumentObject* child: children){
+    for(auto* child: children){
```
Comment (chennes) [inline src/Gui/ViewProviderPyImp.cpp:396]:
```suggestion
    for(auto* child : children){
```

File: src/Gui/ViewProviderPyImp.cpp [hunk 2]
```diff
@@ -393,7 +393,23 @@ PyObject* ViewProviderPy::claimChildren(PyObject* args)
...
+}
+
+PyObject* ViewProviderPy::claimChildrenRecursive(PyObject* args)
+{
+    if (!PyArg_ParseTuple(args, ""))
+        return nullptr;
+
+    std::vector<App::DocumentObject*> children = this->getViewProviderPtr()->claimChildrenRecursive();
+    Py::List ret;
+    for(auto* child: children){
```
Comment (chennes) [inline src/Gui/ViewProviderPyImp.cpp:412]:
```suggestion
    for(auto* child : children){
```

**Review by chennes — APPROVED** [review]
Two linter fixes.

**Comment by PaddleStroke** [comment]
Thanks, lint errors fixed

**Pull request merged**